packages/flutter/test/material/badge_test.dart

```diff
@@ -111,6 +111,63 @@ void main() {
     expect(box, paints..rrect(rrect: RRect.fromLTRBR(-8, -4, 12, 12, const Radius.circular(8)), color: theme.colorScheme.error));
   });
 
+  // Essentially the same as 'Large Badge defaults'
+  testWidgets('Badge.count', (WidgetTester tester) async {
+    late final ThemeData theme;
```

Comment:
Couldn't this just be: 
```suggestion
    late final ThemeData theme = ThemeData.light(useMaterial3: true);
```

And then you wouldn't have to pull it from the context later in the build? You could use it for both the build method and the expect clauses. Or I am missing something with the `Builder` widget?

Reply:
ThemeData.textTheme is updated lazily so the initial value of `ThemeData.light(useMaterial3: true).textTheme.labelSmall` isn't the final value.